core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -75,9 +75,6 @@ void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQuerie
         } else if (sourceQuery instanceof BlendedTermQuery) {
             final BlendedTermQuery blendedTermQuery = (BlendedTermQuery) sourceQuery;
             flatten(blendedTermQuery.rewrite(reader), reader, flatQueries, boost);
-        } else if (sourceQuery instanceof ESToParentBlockJoinQuery) {
-            ESToParentBlockJoinQuery blockJoinQuery = (ESToParentBlockJoinQuery) sourceQuery;
-            flatten(blockJoinQuery.getChildQuery(), reader, flatQueries, boost);
```

Comment:
+1,
This indeed looks similar to the later `else-if` branch which I would prefer for its null check. I only wonder if the order of clauses matters, but I checked BoostingQuery and SynonymQuery and they both don't seem to by supertypes of ESToParentBlockJoinQuery. Maybe @jimczi can check if I missed anything?

Reply:
+1, this is not needed.